### ISSUE TITLE
v3.0.8 might break nextjs 14 (14.1.4) middleware

### ISSUE THREAD

**Issue by danielbush** [sst/open-next#489]
I think v3.0.8 might be breaking my middleware for nextjs 14.1.4 as per this issue I raised in sst: https://github.com/sst/ion/issues/870 .  It might be breaking when a node api is detected by nextjs 14.

In that issue I was getting  a cloudwatch error:

```
WARN	Routing failed. TypeError: globalThis.__import_unsupported is not a function
    at 8915 (file:///var/task/middleware.mjs:17:3341)
    at s (file:///var/task/middleware.mjs:17:1473)
    at 2818 (file:///var/task/middleware.mjs:27:603)
    at s (file:///var/task/middleware.mjs:17:1473)
    at 6872 (file:///var/task/middleware.mjs:27:1308)
    at s (file:///var/task/middleware.mjs:17:1473)
    at 9693 (file:///var/task/middleware.mjs:27:279)
    at s (file:///var/task/middleware.mjs:17:1473)
    at 9108 (file:///var/task/middleware.mjs:25:77770)
    at s (file:///var/task/middleware.mjs:17:1473)
```

If I look at the changes in the release: https://github.com/sst/open-next/compare/v3.0.7...v3.0.8 , the one that seems most germane to the error is the for `packages/open-next/src/build/edge/createEdgeBundle.ts`  which is this commit: https://github.com/sst/open-next/commit/75857cf11906b010ae34c0575e89f9108041556b

I could be wrong.  But I think it's related because it appears to be working around something to do with setting `__import_unsupported` .

I've tested by deploying with open-next@3.0.7 and middleware is nominally working for me in that version. (I tested by setting `openNextVersion: "3.0.7"` in my `sst.config.ts` .

cc @conico974 - don't know if you can verify this somehow?

**Comment by conico974** [comment]
That's very likely where the error is yeah.
The `__import_unsupported` function are supposed to be there to replace module that use node at runtime (to throw an exception). 
If you could create a repro that would help a lot, or at least just find which of your import has a dependency on node so that i can test both with next 15 and 14. 
Recreating the issue is probably harder than the fix itself here.

**Comment by danielbush** [comment]
@conico974 sorry, I'll try to replicate with a vanilla setup.  Incidentally, when I install `sst` using `npm` , the `sst` cli is not present in `node_modules/.bin/sst`.  Although in my actual project there is one.  Do you know how to get a local cli `sst` command installed from an npm package or similar?

**Comment by danielbush** [comment]
seems like `npm i sst@3.0.8` won't set up a cli in node_modules/.bin but `npm i sst` will.

**Comment by danielbush** [comment]
Apologies, I've been struggling to get the time to do the repro.  I can see a possible fix in the above PR 504 .  Happy to test that change if you want (not sure how to get it into my setup)?

**Comment by conico974** [comment]
@danielbush You could do a `pnpm patch` with the fix in the PR (without the `!` as mentionned in the review).
You'll have to install OpenNext and update the `buildCommand` args in the ion construct to use the locally installed OpenNext.

**Comment by conico974** [comment]
@danielbush It should be fixed by #504 in 3.1.2 
If you still have the issue, just post a comment here, i'll reopen the issue
